cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -427,6 +455,21 @@ func (c *Client) AddMember(name string, peerAddrs string) ([]Member, error) {
 	return ret, nil
 }
 
+func memberPromoteWithRety(ctx context.Context, cli *clientv3.Client, learnerID uint64) error {
+	klog.V(1).Infof("[etcd] Promoting a learner as a voting member: %016x", learnerID)
+	err := wait.PollImmediate(15*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
better to tell how the `interval` and `timeout` are defined, even some comments here to tell this is following how `talos` does would be helpful.

Reply:
Added some comments.